pkg/kubelet/volumemanager/volume_manager.go

```diff
@@ -237,11 +240,19 @@ type volumeManager struct {
 	// desiredStateOfWorldPopulator runs an asynchronous periodic loop to
 	// populate the desiredStateOfWorld using the kubelet PodManager.
 	desiredStateOfWorldPopulator populator.DesiredStateOfWorldPopulator
+
+	// the informer factory for CSIDriverLister
+	informerFactory informers.SharedInformerFactory
 }
 
 func (vm *volumeManager) Run(sourcesReady config.SourcesReady, stopCh <-chan struct{}) {
 	defer runtime.HandleCrash()
 
+	// wait for cache sync of CSIDriver
+	if vm.informerFactory != nil {
+		vm.informerFactory.WaitForCacheSync(wait.NeverStop)
```

Comment:
This should pass in the HasSynced for CSIDriver, similar to what's done for AD controller.  That way in the future, if someone adds another object to the shared informer, we don't end up waiting for those objects.

Reply:
Added HasSynced